docs/design/design_vuln_scan_job_in_same_namespace_of_workload.md

```diff
@@ -0,0 +1,81 @@
+# Run Vulnerability Scan Job in Same namespace of workload
+
+## Overview 
+
+Currently, when image is pulled from cloud registries(ecr, acr) in cluster and authentication is set on service account. Then 
+Trivy plugin cannot use that authentication to pull cloud registry image. If user want to get those images scanned then they
+have to specify those credentials at starboard service account([ECR registry configuration]). Also, recently there was one
+option added in Trivy plugin to use Trivy filesystem scan, in which starboard doesnt have option to reuse ImagePullSecret 
+defined on workload spec or on service account. To resolve this limitation, if starboard manages to run scan job in same 
+namespace where workload is running then we can reuse the service account and imagePullSecret available with the workload.
+
+## Solution
+
+Add an environment variable to operator eg `OPERATOR_ENABLE_SCAN_JOB_IN_SAME_NAMESPACE=true`. This way operator can schedule
+and monitor scan jobs in same namespace where workload is running. And plugins will act accordingly to utilize the service 
+account and ImagePullSecret available on the workload.
+
+> :bulb: For this proposal this option will be implemented only for vulnerability scans.
+
+### Example
```

Comment:
Can we add a realistic example base on ECR and IAM where we don't really have imagePullSecret but there's a service account that's annotated in some way to grant it image pull permissions.

Another example might be Trivy FS scanner with imagePullSecret if we consider it.